src/Qir/Runtime/lib/QSharpFoundation/qsharp-foundation-qis.ll

```diff
@@ -31,22 +31,24 @@
 %struct.QirString = type opaque
 %PauliId = type i32
 
-declare void @quantum__rt__message(%"struct.QirString"* %str)
+; TODO: remove this declaration after the https://github.com/microsoft/qsharp-runtime/issues/578 is resolved.
+declare dllimport void @quantum__rt__message(%"struct.QirString"* %str)
```

Comment:
Do we have tests that execute this on Linux? I remember running into an issue with `dllimport` decorations causing a crash on Linux, which was part of the motivation for switching from global variables for `ResultOne` and `ResultZero` to functions (Windows required `dllimport` for it to link, but adding it somehow caused Linux tests to fail at runtime despite `dllimport` supposedly being ignored on non-Windows platforms). I agree with the comment that the real fix here is to resolve #578, but I wouldn't want it to leave us in a state where only Windows works in the meantime.

Reply:
Nice catch!
I double-checked and it looks like the **qsharp_compiler** part of #578 is already fixed (compiler generates `__quantum__rt__message` instead of `__quantum__qis__message__body`).
I just need to remove `__quantum__qis__message__body` and the declaration you pointed at.